src/lib/utils/hashtag.ts

```
@@ -0,0 +1,45 @@
+ */
+export type ContentPart =
+    | { type: 'text'; value: string }
+    | { type: 'hashtag'; value: string };
+
+export function parseContentParts(content: string): ContentPart[] {
+    const parts: ContentPart[] = [];
+    const regex = new RegExp(HASHTAG_PATTERN, 'g');
+    let lastIndex = 0;
+    let match: RegExpExecArray | null;
+
+    while ((match = regex.exec(content)) !== null) {
```

Biome (2.4.13)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)